### ISSUE TITLE
Possible error with embed()

### ISSUE THREAD

**Issue by ernestopye** [lovell/sharp#243]
Unsure if this is on sharp's end, or possibly within libvips. The issue is that when attempting to resize(w,h).embed() an image which has either a width or height lower than w or h respectively, the embed() call simply never gets hit, and you get the full image returned.

For example, with an image that is 200x200, if I do:

resize(100, 150).embed() - works fine, you get a 100x150 image, with vertical padding.
resize(150, 100).embed() - works fine, you get a 150x100 image, with horizontal padding.
resize(200, 150).embed() - works fine, you get a 200x150 image, with horizontal padding (2 param = image width)

However:
resize(201, 150).embed() - returns original image.
resize(150, 201).embed() - returns original image.
resize(220, 220).embed() - returns original image.

This is also noticeable if you have images with much larger widths than heights, say, a logo. If I want to ensure they come back as 200x200, then this returns the original image on most scenarios (h is very small, almost always less than 200, even if the width is generally greater than 200).

Is this expected behavior? I understand we never want to attempt to enlarge an image for obvious reasons. However, since we're padding the images with some background, the issue of quality loss becomes less likely. 

In some of the scenarios I'm running into, there's enough width to the images I'm attempting to resize for them to simply receive vertical padding, but whatever's checking the dimensions seems to cut through and just return the original instead. Even if both the width/height passed to resize are larger than the image's dimensions, could both vertical and horizontal paddings be added?

**Comment by lovell** [comment]
Hi Ernesto, the [embed test cases](https://github.com/lovell/sharp/blob/master/test/unit/embed.js) verify reduction but not enlargement so I suspect it's probably never been supported.

Supporting enlargement would make a useful addition, thank you for the suggestion.

I'd expect padding to apply to only one of either the vertical or horizontal edges. Padding on all sides sounds more like #128.

> "I understand we never want to attempt to enlarge an image for obvious reasons."

Only when using the `withoutEnlargement` option, I think. We might need to test the interaction of an attempted enlargement+embed with this option as well as `max`and `min`.

I'd be very happy to accept a [contribution](https://github.com/lovell/sharp/blob/master/CONTRIBUTING.md#submit-a-pull-request-with-a-new-feature) implementing some or all of this feature if you're in a position to help. It looks like you've already got plenty of test cases!

**Comment by ernestopye** [comment]
Absolutely. I've forked the repo and will try to take a closer look this week.

**Comment by lovell** [comment]
Hi @ernestopye, sorry for the delay in responding. Do you think the addition of a `pad` feature, as described by the comments in #128, might provide enough API to solve the problem you're facing?

**Comment by ernestopye** [comment]
Hey @lovell, no worries. Yes, I think it would as long as withoutEnlargement = true was handled in a way that avoids the scenario I'm describing above. It would basically never prevent resizing/padding of an image, even if both the width/height of the original image are lower than the target width/height, as padding could be added around the whole thing.
